Some general cleanup, edits, and expanded hello_world/installation pages.

eX-n0x opened the pull request:
### About this PR:
Attempts to freshen up the document as well as adding critical information and cleaning linting errors that could cause issues with parsers.

### Why is this needed:
_bababooey_

### Changelog:
- added installation guide for byond
- added explanation on how to use dream maker in hello_world
- fixed numerous linting issues
- fixed some links
- various grammar edits
- slightly modified some page styles
- other stuff maybe

eX-n0x committed:
- added installation for byond
- added explanation on how to use byond editor (dream maker)
- fixed numerous linting issues
- fixed some links
- various grammar edits
- slightly modified some page styles
- other stuff maybe

ZeWaka commented on src/SUMMARY.md:
```diff
@@ -4,40 +4,40 @@
 
 - [Hello World](./hello_world.md)
-	- [Comments](./hello/comments.md)
-	- [Basic Types](./hello/types.md)
```
hard tabs, not spaces.

ZeWaka commented on src/hello/comments.md:
```diff
@@ -3,12 +3,12 @@
 
 * Regular comments which are ignored by the compiler:
-	* `// Line comments which go to the end of the line.`
-	* `/* Block comments which go to the closing delimiter. */`
```
hard tabs, not spaces.

ZeWaka commented on src/preprocessor.md:
```diff
@@ -1,21 +1,25 @@
+Sometimes, you'll want to have shared values across your project instead of copy-pasting them everywhere. This is where `#define` comes in.
 
-The `#define Name Value` statement substitues the `Name` for `Value` for as long as the define is not `#undef`ined. Substitution only applies to whole words. Text inside of double or single quotes is not processed for substitution, so `"This is BIG."` would not be modified even if a define named `BIG` were defined. That is different from `"This is [BIG]."`, where `BIG` is an embedded expression,  ...
+The `#define Name Value` statement substitues the `Name` for `Value` as long as the define is not `#undef`-ined. Substitution only applies to whole words. Text inside of double or single quotes is not processed for substitution, so `"This is BIG."` would not be modified even if a define named `BIG` were to be defined. That's different from `"This is [BIG]."`, whereas `BIG` is an embedded expressi ...
```
```suggestion
The `#define Name Value` statement substitues the `Name` for `Value` as long for as the define is not `#undef`-ined. Substitution only applies to whole words. Text inside of double or single quotes is not processed for substitution, so `"This is BIG."` would not be modified even if a define named `BIG` were to be defined. That's different from `"This is [BIG]."`, whereas `BIG` is an embedded expression, which does get processed for macro substitution.
```
the `for` is intentional.

ZeWaka commented on src/ops/bitflags.md:
```diff
@@ -27,5 +27,6 @@ Since DM only supports floating point numbers, we are restricted to 24 bits (23
-> Hint: What other binary operators are there?
+
+>Using the above code snippet as a base, try implementing the ability to toggle a flag easily.
+>>**Hint**: What other binary operators are there?
```
is this valid syntax? GitHub doesn't think so.

eX-n0x replied:
Should be, both the VSCode preview and GitHub style preview 
<img width="179" alt="Screenshot 2021-06-16 182926" src="https://user-images.githubusercontent.com/45720148/122307933-f2629b00-ced0-11eb-99b7-8babdcc3359d.png">
render it without issues.

ZeWaka commented on src/objs/disposal.md:
```diff
@@ -15,19 +17,23 @@ Del(foo)
 The garbage collector works by using a reference counting system. Once an object is no longer referenced, it gets deleted.
 
-Important to note: circular references will never be deleted by the garbage collector. This is defined as a pair of objects with variables that point to each other, or even an object with a variable that points to itself. Also, an object with running or sleeping procs will not be deleted.
+>**Important to Note**: Circular references will never be deleted by the garbage collector. This is defined as a pair of objects with variables that point to each other, or even an object with a variable that points to itself. Also, an object with running or sleeping procs will not be deleted.
```
```suggestion
>**Important to note**: Circular references will never be deleted by the garbage collector. This is defined as a pair of objects with variables that point to each other, or even an object with a variable that points to itself. Also, an object with running or sleeping procs will not be deleted.
```
why capitalize

ZeWaka commented on src/hello_world.md:
```diff
@@ -14,15 +16,23 @@ This is the source code of the traditional Hello World program.
 `world << "x"` is the most basic way to output text to the default window, using the << [operator].
 
-To run this, simply open up a blank byond project, and compile it.
+## Starting Dream Maker
```
I would prefer the guide not use Dream Maker unless necessary - see sprites.

eX-n0x committed:
* adjust vscode settings to better match project intentions
- removed spaces, added back hard tabs
- removed Dream Maker guide
- reverted some minor changes
